packages/next/server/on-demand-entry-handler.js

```diff
@@ -234,7 +234,11 @@ export default function onDemandEntryHandler (devMiddleware, multiCompiler, {
       const absolutePagePath = pagePath.startsWith('next/dist/pages') ? require.resolve(pagePath) : join(pagesDir, pagePath)
 
       page = posix.normalize(pageUrl)
-      const result = { isAmp, pathname: page }
+      const result = {
+        isAmp,
+        pathname: page,
+        [isAmp ? 'hasCanonical' : 'hasAmp']: Boolean(await findPageFile(pagesDir, normalizedPagePath, pageExtensions, !isAmp, ampEnabled))
```

Comment:
Does this still need the try {} catch {}?

Reply:
`findPageFile` doesn't throw